src/utils/roll-orb.ts

```diff
@@ -0,0 +1,159 @@
+  const branchName = `roller/orb/${orbTarget.name}/${MAIN_BRANCH}`;
+  const shortRef = `heads/${branchName}`;
+  const ref = `refs/${shortRef}`;
+  const { owner, repo } = repository;
+
+  const { data } = await github.repos.getContent({
+    owner,
+    repo,
+    path: filePath,
+    ref: MAIN_BRANCH,
+  });
+
+  // Look for a pre-existing PR that targets this branch to see if we can update that.
+  const existingPrsForBranch = (await github.paginate('GET /repos/:owner/:repo/pulls', {
+    head: branchName,
+    ...orbTarget,
+    state: 'open',
+  })) as PullsListResponseItem[];
+
+  const prs = existingPrsForBranch.filter(pr =>
+    pr.title.startsWith(`chore: bump ${orbTarget.name}`),
+  );
+
+  if (prs.length) {
+    // Update existing PR(s)
+    for (const pr of prs) {
+      d(`Found existing PR: #${pr.number} opened by ${pr.user.login}`);
+
+      // Check to see if automatic orb roll has been temporarily disabled
+      const hasPauseLabel = pr.labels.some(label => label.name === 'roller/pause');
+      if (hasPauseLabel) {
+        d(`Automatic updates have been paused for #${pr.number}, skipping orb roll.`);
+        continue;
+      }
+
+      d(`Attempting orb update for #${pr.number}`);
+
+      const previousVersion = await updateConfigFile(data, orbTarget, targetValue);
+
+      if (previousVersion === targetValue) {
+        d(`orb version unchanged - skipping PR body update`);
+        continue;
+      }
+
+      d(`orb version changed - updating PR body`);
+
+      const re = new RegExp('^Original-Version: (\\S+)', 'm');
+      const prVersionText = re.exec(pr.body);
+
+      if (!prVersionText || prVersionText.length === 0) {
+        d('Could not find PR version text in existing PR - exiting');
+        return;
+      }
+
+      await github.pulls.update({
+        owner,
+        repo,
+        pull_number: pr.number,
+        ...getOrbPRText(orbTarget, {
+          previousVersion,
+          newVersion: targetValue,
+          branchName: MAIN_BRANCH,
+        }),
+      });
+    }
+  } else {
+    try {
+      d(`roll triggered for  ${orbTarget.name}=${targetValue}`);
+
+      if (!('content' in data)) return;
+
+      try {
+        await github.git.getRef({ owner, repo, ref: shortRef });
+        d(`Ref ${ref} already exists`);
+      } catch {
+        d(`Creating ref=${ref} at sha=${sha}`);
+        await github.git.createRef({ owner, repo, ref, sha });
+      }
+
+      const previousVersion = await updateConfigFile(data, orbTarget, targetValue);
+
+      d(`Raising a PR for ${branchName} to ${repo}`);
+      await github.pulls.create({
+        ...repository,
+        base: MAIN_BRANCH,
+        head: `${owner}:${branchName}`,
+        ...getOrbPRText(orbTarget, {
+          previousVersion,
+          newVersion: targetValue,
+          branchName: MAIN_BRANCH,
+        }),
+      });
+    } catch (e) {
+      d(`Error rolling ${repository.owner}/${repository.repo} to ${targetValue}`, e);
+    }
+  }
```

Comment:
Won't this always create the branch with `github.git.createRef`? It's creating the ref before checking if `updateConfigFile` skipped.